receiver/awscloudwatchreceiver/logs.go

```diff
@@ -274,13 +276,12 @@ func (l *logsReceiver) discoverGroups(ctx context.Context, auto *AutodiscoverCon
 	numGroups := 0
 	var nextToken = aws.String("")
 	for nextToken != nil {
-		if numGroups > auto.Limit {
+		if numGroups >= auto.Limit {
```

Comment:
is `auto.Limit` still configurable?

Reply:
Yes is still configurable, I believe the request fails validation if you field a limit value > 50. I think users should still be able to configure a value greater than 50.

Reply:
It feels weird to me to allow the value to be configured but then use the `maxLogGroupsPerDiscovery` constant, but I admit that I am not familiar with this component.  Is it beneficial to allow the limit to be configurable, but in the config's `Validate` function we enforce a max of 50?  What is the reason to allow a value greater than 50 to be configured?

Reply:
We actually don't enforce a max of 50 in the config's validate function.

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/ad5e5bbc42b3095ef34de9f71df472a5ac21325b/receiver/awscloudwatchreceiver/config.go#L138-L142

The limit in the config for autodiscovery is limiting how many log groups are discovered via autodiscovery. Cloudwatch logs are [hierarchical](https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/Working-with-log-groups-and-streams.html).

```
-  region (us-west-2)
-- Log Group 1 (i.e. `/aws/eks/dev-0/`)
--- Log Stream 1 
--- Log Stream 2
-- Log Group 2 (i.e. `/aws/eks/dev-1`)
--- Log Stream 1
--- Log Stream 2
```

Say a new dev cluster gets added and you would like to discover further clusters like a `dev-{2...n}`. A limit is a nicety to have in case too many clusters/lambda functions/deployments get spun up that match a prefix and are discovered. Having a safeguard against this (and ultimately being the one who foots the bill) I think is warranted.

Overall I think this change was to fix the issue where the API rejects requests that are asking for more than 50 Log Groups at a time which is an API limitation not something that blocks the receiver from obtaining this information.

Reply:
Thanks for the explanation.  My naive reasoning is that if this receiver calls the API to collect data and the API has a limitation in how it its request is configured, then this receiver should also have that limitation in its configuration.

If the receiver allows the user sets the limit to be 100, as a user I would expect the receiver to use that value.  It looks like we're allowing the value to be configured but use 50 in the actual request no matter what.  As a user I'd find that confusing.  Let me know if I'm still misunderstanding something.

Reply:
> If the receiver allows the user sets the limit to be 100, as a user I would expect the receiver to use that value. It looks like we're allowing the value to be configured but use 50 in the actual request no matter what. As a user I'd find that confusing. Let me know if I'm still misunderstanding something.

The limit is being used, just as a break condition. As described in the README is `Limits the number of discovered log groups.` 

The way it would flow for your use case of a limit of 100 would be

```
receiver start

receiver is configured for autodiscovery, makes first request (it cannot request the whole 100, it needs multiple requests)

gets response with 50 log groups discovered (total number of groups discovered is 50)

makes 2nd request 

gets response with 50 log groups (total is now 100)

reached configured limit, break out of making discovery requests
```

In the case of having a limit of non-50 increments it would

For example `limit: 75` but 100 match prefix:

```
make 1st request and get 50 
make 2nd request and get 50 if more matched the prefix 
on the 75th we break because it equal to our limit and we don't collect the next 25
```

I can add another config param that is like `discovery_request_size` but I cannot justify its existence as to why someone would ever want less than the max. Each request to the Cloudwatch API comes with a charge, so I picture it being the most cost-beneficial to not make more requests than necessary.

Can you think of a good reason to have that value be configurable for a user?

Reply:
Ah ok, I see now how limit is being used outside the API call itself.  I agree we don't need a separate configuration option for the actual API request limit.  

Can you update something in the README that specifies `autodiscover.limit` is a restrictor for how many groups the receiver will discover but not how large each API batch will be?  It looks like `auto.Limit` used to do both, but now we're restricting it to just be how many the collector is receive.

Thanks for walking me through this!

Reply:
I added to the docs about this, thanks for the question! 🚀 

Happy to explain any other thing further!